### PR TITLE
added notebooks for #9656, #9737 and tests for annotation addition and deletion

### PR DESCRIPTION
**Pull request by iftwigs**

### CODE REVIEW

**Comment by da-konfuzio** [comment]
At notebooks/page_split_training_model_NBOW.ipynb

Can you add documentation explaining the motivation behind our custom metric at `def calculate_metrics`?

Probably it's not immediately clear to someone reading this that we are measuring correct page splits at the location of each predicted FIRST_PAGE.

File: notebooks/page_split_training_model_BERT.ipynb [hunk 1]
```diff
@@ -0,0 +1,6750 @@
...
+    },
+    "id": "6mWpSC5Kafs_",
+    "outputId": "d2e48985-9d7d-4b50-ddd5-19d9ccae5da2"
+   },
+   "outputs": [],
+   "source": [
+    "dataset = load_dataset('csv',\n",
+    "                      data_files={'train': 'drive/MyDrive/knfz/train_1644.csv',\n",
+    "                                 'test': 'drive/MyDrive/knfz/test_1644.csv'})"
+   ]
```
Comment (da-konfuzio) [inline notebooks/page_split_training_model_BERT.ipynb:280]: In the NBOW notebook you built the dataset by using the SDK Project data. Here it seems you did the same but saved the results in a CSV in your Drive and removed the part of the notebook that generated the CSV.

Can you please include that code back in, and remove references to your Drive?

File: notebooks/page_split_training_model_BERT.ipynb [hunk 2]
```diff
@@ -0,0 +1,6750 @@
...
+     "output_type": "stream",
+     "text": [
+      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: Unnamed: 0, text. If Unnamed: 0, text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
+      "***** Running Evaluation *****\n",
+      "  Num examples = 235\n",
+      "  Batch size = 4\n",
+      "Saving model checkpoint to drive/MyDrive/knfz/model/checkpoint-1000\n",
+      "Configuration saved in drive/MyDrive/knfz/model/checkpoint-1000/config.json\n",
+      "Model weights saved in drive/MyDrive/knfz/model/checkpoint-1000/pytorch_model.bin\n",
+      "tokenizer config file saved in drive/MyDrive/knfz/model/checkpoint-1000/tokenizer_config.json\n",
```
Comment (da-konfuzio) [inline notebooks/page_split_training_model_BERT.ipynb:667]: Please remove references to external drives and save the models to the local folder

File: notebooks/page_split_training_model_BERT.ipynb [hunk 3]
```diff
@@ -0,0 +1,6750 @@
...
+      "Special tokens file saved in drive/MyDrive/knfz/model/checkpoint-1000/special_tokens_map.json\n",
+      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: Unnamed: 0, text. If Unnamed: 0, text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
+      "***** Running Evaluation *****\n",
+      "  Num examples = 235\n",
+      "  Batch size = 4\n",
+      "Saving model checkpoint to drive/MyDrive/knfz/model/checkpoint-2000\n",
+      "Configuration saved in drive/MyDrive/knfz/model/checkpoint-2000/config.json\n",
+      "Model weights saved in drive/MyDrive/knfz/model/checkpoint-2000/pytorch_model.bin\n",
+      "tokenizer config file saved in drive/MyDrive/knfz/model/checkpoint-2000/tokenizer_config.json\n",
+      "Special tokens file saved in drive/MyDrive/knfz/model/checkpoint-2000/special_tokens_map.json\n",
```
Comment (da-konfuzio) [inline notebooks/page_split_training_model_BERT.ipynb:677]: See above

File: notebooks/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+datasets==2.4.0
+keras==2.9.0
+Keras-Preprocessing==1.1.2
+nltk==3.6.3
```
Comment (da-konfuzio) [inline notebooks/requirements.txt:4]: Is this nltk version an exact requirement or should it be nltk>=3.6.3?

**Comment by da-konfuzio** [comment]
At notebooks/page_split_training_model_NBOW.ipynb

About word_tokenize, have you checked for size limits of the accepted text, and what can happen in such edge cases (we can have very large pages)?

**Comment by da-konfuzio** [comment]
About the keras.preprocessing.text.Tokenizer (https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/text/Tokenizer)

With the `filters` argument it's possible to filter out particular kinds of data  from the page text (such as numbers or dates). It's very likely that by playing around with this we can improve performance.

Similarly with the `split` argument, playing around could turn out to be useful (e.g. separate by sentence or by line breaks instead of by whitespace https://stackoverflow.com/a/63246520/5800596).

**Comment by da-konfuzio** [comment]
About the word_tokenize, some experimentation should be done by removing stopwords from the vocabulary, and see how scores change (see https://stackabuse.com/removing-stop-words-from-strings-in-python/)
